GliaWidgets/Sources/Layout/Layout+UIView.swift

```diff
@@ -0,0 +1,251 @@
+    func layoutInSuperviewCenter(
+        edges: Align = .center,
+        xOffset: CGFloat = 0.0,
+        yOffset: CGFloat = 0.0
+    ) -> [NSLayoutConstraint] {
+        guard let superview = superview else {
+            assertionFailure("Can't layout view due to it is not added to any view hierarcy.")
+            return []
+        }
+
+        return layoutInCenter(
+            superview,
+            edges: edges,
+            xOffset: xOffset,
+            yOffset: yOffset
+        )
+    }
+
+    /// Creates constraints to layout the current view in the center of passed guide (ex.: superview,
+    ///  safeAreaLayoutGuide, etc.). Created constraints rely only on `centerXAnchor` and `centerYAnchor` anchors.
+    /// - Returns: Array of constraints.
+    func layoutInCenter(
+        _ guide: Layoutable,
+        edges: Align = .center,
+        xOffset: CGFloat = 0.0,
+        yOffset: CGFloat = 0.0
+    ) -> [NSLayoutConstraint] {
+        translatesAutoresizingMaskIntoConstraints = false
+
+        var constraints = [NSLayoutConstraint]()
+
+        if edges.contains(.horizontal) {
+            constraints += centerXAnchor.constraint(equalTo: guide.centerXAnchor, constant: xOffset)
+        }
+
+        if edges.contains(.vertical) {
+            constraints += centerYAnchor.constraint(equalTo: guide.centerYAnchor, constant: yOffset)
+        }
+
+        return constraints
+    }
+
+    /// Creates constraints to layout the current view in the superview and rely on passed parameters `edges` and
+    ///  `insets`. Created constraints rely on `topAnchor`, `bottomAnchor`, `leadingAnchor`, `trailingAnchor` anchors
+    ///  and doesn't rely on `widhtAnchor`, `heightAnchor`.
+    /// - Returns: Array of constraints or empty array, if the current view doesn't have superview.
+    func layoutInSuperview(
+        edges: Edge = .all,
+        insets: UIEdgeInsets = .zero,
+        priority: UILayoutPriority = .required
+    ) -> [NSLayoutConstraint] {
+        guard let superview = superview else {
+            assertionFailure("Can't layout view due to it is not added to any view hierarcy.")
+            return [NSLayoutConstraint]()
+        }
+
+        return layoutIn(
+            superview,
+            edges: edges,
+            insets: insets,
+            priority: priority
+        )
+    }
+
+    /// Create constraints to layout the current view in the scroll view `edges` and  `insets`. Created constraints rely
+    ///  on `topAnchor`, `leadingAnchor`, `trailingAnchor` anchors, and don't rely on `bottomAnchor` (since it is
+    ///  layouting inside scroll view), `widhtAnchor`, `heightAnchor`.
+    /// - Returns: Array of constraints or empty array, if the current view doesn't have superview.
+    func layoutInScrollView(
+        edges: Edge = .all,
+        insets: UIEdgeInsets = .zero,
+        priority: UILayoutPriority = .required
+    ) -> [NSLayoutConstraint] {
+        guard superview is UIScrollView else {
```

Comment:
I am not sure about the fact that if the superview is not a scroll view, then it doesn't fail.

Reply:
fixed - a4a731726fb8919f6066d8435fcac255e72d57c8
simplified `layoutInSuperview` function to avoid explicit differentiation between UIScrollView and UIView